tests/integration/generic_test.go

```diff
@@ -331,8 +331,26 @@ var _ = Describe("odo generic", func() {
 		It("should not allow creating a URL with long name", func() {
 			helper.CmdShouldPass("odo", "create", "nodejs", "--project", project)
 			stdOut := helper.CmdShouldFail("odo", "url", "create", testLongURLName, "--port", "8080")
-			Expect(stdOut).To(ContainSubstring("url name must be shorter than 63 characters"))
-		})
+			Expect(stdOut).To(ContainSubstring("must be shorter than 63 characters"))
+		})
+		/*
+			// these tests only work when experimental mode is activated
+			It("should not allow creating an invalid host", func() {
+				helper.CmdShouldPass("odo", "create", "nodejs", "--project", project)
+				stdOut := helper.CmdShouldFail("odo", "url", "create", "--host", "https://127.0.0.1:60104", "--ingress")
+				Expect(stdOut).To(ContainSubstring("is not a valid host name"))
+			})
+			It("should not allow using tls secret if url is not secure", func() {
+				helper.CmdShouldPass("odo", "create", "nodejs", "--project", project)
+				stdOut := helper.CmdShouldFail("odo", "url", "create", "--tls-secret", "foo", "--ingress")
+				Expect(stdOut).To(ContainSubstring("TLS secret is only available for secure URLs of Ingress kind"))
+			})
+				It("should report multiple issues when it's the case", func() {
+					helper.CmdShouldPass("odo", "create", "nodejs", "--project", project)
+					stdOut := helper.CmdShouldFail("odo", "url", "create", "--host", "https://127.0.0.1:60104", "--tls-secret", "foo", "--ingress")
+					Expect(stdOut).To(And(ContainSubstring("is not a valid host name"), ContainSubstring("TLS secret is only available for secure URLs of Ingress kind")))
+				})
+		*/
```

Comment:
These tests should be added https://github.com/openshift/odo/blob/master/tests/integration/devfile/cmd_devfile_url_test.go. All the tests there run for devfile URLs with experimental mode turned on.

Reply:
Thank you, I'm not familiar with the tests' organization.